jctools-core/src/main/java/org/jctools/queues/MpmcUnboundedXaddArrayQueue.java

```diff
@@ -658,7 +658,97 @@ public boolean relaxedOffer(E e)
     @Override
     public E relaxedPoll()
     {
-        return poll();
+        final int chunkMask = this.chunkMask;
+        final int chunkShift = this.chunkShift;
+        final int chunkSize = chunkMask + 1;
+        final long consumerIndex = this.lvConsumerIndex();
+        final AtomicChunk<E> consumerBuffer = this.lvConsumerBuffer();
+        final int consumerOffset = (int) (consumerIndex & chunkMask);
+        final long chunkIndex = consumerIndex >> chunkShift;
+        final boolean firstElementOfNewChunk = consumerOffset == 0 && consumerIndex >= chunkSize;
+        if (firstElementOfNewChunk)
+        {
+            final AtomicChunk<E> next = consumerBuffer.lvNext();
+            final long expectedChunkIndex = chunkIndex - 1;
+            if (expectedChunkIndex != consumerBuffer.lvIndex() || next == null)
+            {
+                return null;
+            }
+            E e = null;
+            final boolean pooled = next.isPooled();
+            if (pooled)
+            {
+                if (next.lvSequence(consumerOffset) != chunkIndex)
+                {
+                    return null;
+                }
+            }
+            else
+            {
+                e = next.lvElement(consumerOffset);
```

Comment:
@nitsanw We don't know here if `next::index == chunkIndex`, but we don't care as long as `e !=null` and we succeed to increment `consumerIndex`